source/tutorials/mirror-upstream-server.rst

```diff
@@ -0,0 +1,231 @@
+.. _mirror-upstream-server:
+  
+Set Up a Mirror of the |CL| Upstream Update Server
+##################################################
+
+This tutorial shows how to set up a mirror of the upstream |CL| server
+and configure your |CL| clients to use it.
+
+.. contents::
+    :local:
+    :depth: 1
+
+Prerequisites
+*************
+
+* The recommended disk space for the mirror server should have at least 100GB 
+  of disk space as each complete update content is approximately 45GB.  
+
+Install up |CL| server to host updates
+**************************************
+
+#. Follow the :ref:`bare-metal-install-server` guide to install |CL| server.
+   Add a user with `Administrator` privilege.
+
+#. After installation is complete, boot it up.
+
+#. Add the `wget` bundle.  This will be used to clone the upstream |CL| server.  
+
+   .. code-block:: bash
+
+      sudo swupd bundle-add wget
+
+Clone the |CL| update content
+*****************************
+
+|CL| periodically releases a "minversion", which is a complete update.  
+Then, subsequent releases are small updates until the next minversion.
+
+Download a minversion to start your mirror.  
+
+#. Determine a proper minversion by looking at a :file:`Manifest.Mom` file 
+   for a minversion of |CL|.
+
+   For example, if you look at 
+   https://cdn.download.clearlinux.org/update/33010/Manifest.MoM, 
+   you will see that the minversion is 32900.  So clone this version as the 
+   starting point.
+
+   .. code-block:: console
+      :emphasize-lines: 4
+
+      MANIFEST        30
+      version:        33010
+      previous:       33000
+      minversion:     32900
+      filecount:      1131
+      timestamp:      1588358889
+      contentsize:    0
+
+#. Make a directory to store the cache.
+
+   .. code-block:: bash
+
+      mkdir ~/mirror-download-clearlinux-org && cd $_
+
+#. Download the “0” folder.
+
+   .. code-block:: bash
+
+      wget --no-verbose \
+      --no-parent --recursive \
+      --no-host-directories -erobots=off \
+      --reject "index.html" https://cdn.download.clearlinux.org/update/0/
+
+#. Download the “version” folder.
```

Comment:
I changed it to "Recursively download the :file:`update/version` folder."  This is more explicit.  I didn't say it will take some time because it's very small (~632K) and take less than a minute.